api/soap/mc_issue_api.php

```diff
@@ -90,7 +90,243 @@ function mc_issue_get( $p_username, $p_password, $p_issue_id ) {
+			case BUGNOTE_UPDATED:
+			case BUGNOTE_DELETED:
+				$t_show_new_value = false;
+				$t_old_value_name = 'note';
+				break;
+			case BUGNOTE_STATE_CHANGED:
+				$t_old_value_name = 'view_state';
+				$t_new_value_name = 'note';
+				break;
+			case BUG_ADD_RELATIONSHIP:
+			case BUG_REPLACE_RELATIONSHIP:
+			case BUG_DEL_RELATIONSHIP:
+				$t_old_value_name = 'relationship';
+				$t_new_value_name = 'issue';
+				break;
+			case BUG_CLONED_TO:
+				$t_show_old_value = false;
+				$t_new_value_name = 'issue';
+				break;
+		}
+
+		$t_files = file_get_visible_attachments( $p_issue_id );
+
+		$fn_process_value = function( $p_issue_id, $p_type, $p_field, $p_value, $p_lang, $p_new_value ) use ( $t_files ) {
+			if( is_blank( $p_value ) ) {
+				return '';
+			}
+
+			switch( $p_type ) {
+				case TAG_ATTACHED:
+				case TAG_DETACHED:
+					$t_tag = tag_get_by_name( $p_value );
+					if( $t_tag === false ) {
+						return array( 'name' => $p_value );
+					}
+
+					return array( 'id' => $t_tag['id'], 'name' => $t_tag['name'] );
+				case BUGNOTE_ADDED:
+				case BUGNOTE_DELETED:
+					return array( 'id' => (int)$p_value );
+				case BUGNOTE_UPDATED:
+					if( !$p_new_value ) {
+						return array( 'id' => (int) $p_value );
+					}
+
+					return '';
+				case FILE_ADDED:
+				case FILE_DELETED:
+					$t_value = array();
+
+					$t_id = null;
+					foreach( $t_files as $t_file ) {
+						if( $t_file['display_name'] == $p_value ) {
+							$t_value['id'] = (int)$t_file['id'];
+							break;
+						}
+					}
+
+					$t_value['filename'] = $p_value;
+
+					return $t_value;
+				case BUGNOTE_STATE_CHANGED:
+					if( $p_new_value ) {
+						return array( 'id' => (int)$p_value );
+					}
+
+					return mci_enum_get_array_by_id( (int)$p_value, 'view_state', $p_lang );
+				case BUG_ADD_RELATIONSHIP:
+				case BUG_REPLACE_RELATIONSHIP:
+				case BUG_DEL_RELATIONSHIP:
+					if( $p_new_value ) {
+						return array( 'id' => (int)$p_value );
+					}
+
+					return array(
+						'id' => (int)$p_value,
+						'name' => relationship_get_name_for_api( (int)$p_value ),
+						'label' => relationship_get_description_for_history( (int)$p_value ) );
+				case BUG_CLONED_TO:
+					return array( 'id' => (int)$p_value );
+			}
+
+			switch( $p_field ) {
+				case 'status':
+					$t_value = mci_enum_get_array_by_id( (int)$p_value, 'status', $p_lang );
+					break;
+				case 'priority':
+					$t_value = mci_enum_get_array_by_id( (int)$p_value, 'priority', $p_lang );
+					break;
+				case 'severity':
+					$t_value = mci_enum_get_array_by_id( (int)$p_value, 'severity', $p_lang );
+					break;
+				case 'view_state':
+					$t_value = mci_enum_get_array_by_id( (int)$p_value, 'view_state', $p_lang );
+					break;
+				case 'resolution':
+					$t_value = mci_enum_get_array_by_id( (int)$p_value, 'resolution', $p_lang );
+					break;
+				case 'reproducibility':
+					$t_value = mci_enum_get_array_by_id( (int)$p_value, 'reproducibility', $p_lang );
+					break;
```

Comment:
What about
```php
				case 'status':
				case 'priority':
				case 'severity':
				case 'view_state':
				case 'resolution':
				case 'reproducibility':
					$t_value = mci_enum_get_array_by_id( (int)$p_value, $p_field, $p_lang );
					break;
```

Reply:
I would rather keep it as is.  The idea is that we should be moving to constants for DB fields and constants for enums and in such case having such shortcuts will have to be re-worked then.